### PR TITLE
quickfort: expose basic buildingplan functionality to Lua

### PR DESCRIPTION
**Pull request by myk002**
#499 
More changes are coming to support more buildings and better filtering, but I wanted to submit this as a separate PR so we could discuss any general concerns you might have about extending buildingplan functionality.

### CODE REVIEW

File: plugins/buildingplan-lib.cpp [hunk 1]
```diff
@@ -623,11 +623,6 @@ bool Planner::allocatePlannedBuilding(df::building_type type)
         return false;
     }
 
-    for (auto iter = newinst->jobs.begin(); iter != newinst->jobs.end(); iter++)
-    {
-        (*iter)->flags.bits.suspend = true;
```
Comment (lethosor) [inline plugins/buildingplan-lib.cpp:628]: Out of curiosity, why did this move to addPlannedBuilding?

File: plugins/lua/buildingplan.lua [hunk 2]
```diff
@@ -0,0 +1,3 @@
+local _ENV = mkmodule('plugins.buildingplan')
+
```
Comment (lethosor) [inline plugins/lua/buildingplan.lua:2]: Lua API docs would be great - simple is fine for these functions, just as long as people know that the functions exist. Listing them in `Lua API.rst` is probably more important than listing them here, but seeing this reminded me. (If you wanted to wait for a  confirmation before documenting these functions, they look sensible to me.)

File: plugins/buildingplan-lib.cpp (hunk 1 above)
Comment (myk002) [reply plugins/buildingplan-lib.cpp:628]: addPlannedBuilding is an API method, but it didn't seem to enforce the supposed invariant that planned buildings are suspended. Refactoring the functionality into the method itself seemed cleaner than making it a separate API method or reimplementing the suspension in Lua.

File: plugins/lua/buildingplan.lua (hunk 2 above)
Comment (myk002) [reply plugins/lua/buildingplan.lua:2]: done. thanks!

File: plugins/buildingplan-lib.cpp (hunk 1 above)
Comment (lethosor) [reply plugins/buildingplan-lib.cpp:628]: Ah, I see allocatePlannedBuilding calling addPlannedBuilding now (the door check was between this removed section and that call, but didn't need to be).

**Pull request merged**